cli/repair/clusterwide_config.go

```diff
@@ -0,0 +1,302 @@
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get instances config: %s", err)
+	}
+	topologyConf.Instances = *instancesConf
+
+	replicasetsConf, err := getReplicasetsConf(&topologyConf)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get replicasets config: %s", err)
+	}
+	topologyConf.Replicasets = *replicasetsConf
+
+	return &topologyConf, nil
+}
+
+// INSTANCES
+
+func getInstancesConf(topologyConf *TopologyConfType) (*map[string]InstanceConfType, error) {
+	instancesConfRaw, found := topologyConf.Raw[keyInstances]
+	if !found {
+		return nil, fmt.Errorf("Topology config doesn't contain %q key", keyInstances)
+	}
+
+	instancesConfRawMap, ok := instancesConfRaw.(RawConfType)
+	if !ok {
+		return nil, fmt.Errorf("%q value isn't a map", keyInstances)
+	}
+
+	topologyConf.InstancesRaw = instancesConfRawMap
+
+	instancesConf := make(map[string]InstanceConfType)
+
+	for instanceUUIDRaw, instanceConfRaw := range instancesConfRawMap {
+		instanceUUID, ok := instanceUUIDRaw.(string)
+		if !ok {
+			return nil, fmt.Errorf("Instance UUID isn't a string")
+		}
+
+		var instanceConf InstanceConfType
+
+		switch conf := instanceConfRaw.(type) {
+		case string:
+			if conf != expelledState {
+				return nil, fmt.Errorf("Instance %s is in the unknown state %s", instanceUUID, conf)
+			}
+			instanceConf.IsExpelled = true
+		case RawConfType:
+			isDisabled, ok := conf[keyInstanceDisabled]
+			if !ok {
+				return nil, fmt.Errorf("Instance %s config doesn't contain %q key", instanceUUID, keyInstanceDisabled)
+			}
+			instanceConf.IsDisabled, ok = isDisabled.(bool)
+			if !ok {
+				return nil, fmt.Errorf("Instance %s has %q that isn't a bool", instanceUUID, keyInstanceDisabled)
+			}
+
+			advertiseURI, ok := conf[keyInstanceAdvertiseURI]
+			if !ok {
+				return nil, fmt.Errorf("Instance %s config doesn't contain %q key", instanceUUID, keyInstanceAdvertiseURI)
+			}
+			instanceConf.AdvertiseURI, ok = advertiseURI.(string)
+			if !ok {
+				return nil, fmt.Errorf("Instance %s has %q that isn't a string", instanceUUID, keyInstanceAdvertiseURI)
+			}
+
+			replicasetUUID, ok := conf[keyInstanceReplicasetUUID]
+			if !ok {
+				return nil, fmt.Errorf("Instance %s config doesn't contain %q key", instanceUUID, keyInstanceReplicasetUUID)
+			}
+			instanceConf.ReplicasetUUID, ok = replicasetUUID.(string)
+			if !ok {
+				return nil, fmt.Errorf("Instance %s has %q that isn't a string", instanceUUID, keyInstanceReplicasetUUID)
+			}
+
+			instanceConf.Raw = conf
+		default:
+			return nil, fmt.Errorf("Instance %s config isn't a map or a string", instanceUUID)
+		}
+
+		instancesConf[instanceUUID] = instanceConf
+	}
+
+	return &instancesConf, nil
+}
+
+func setInstanceURIRaw(topologyConf *TopologyConfType, instanceUUID, newURI string) error {
+	instanceConf, ok := topologyConf.Instances[instanceUUID]
+	if !ok {
+		return fmt.Errorf("Instance %s isn't found in cluster", instanceUUID)
+	}
+
+	if instanceConf.IsExpelled {
```

Comment:
I this we should introduce status name, not flags "isExpelled"/"isDisabled" etc. It allows to add new statuses when it will be introduced.

Reply:
This statuses are processed in a very different way, I don't want to compare status with a constant, it's easier to use flag.